services/QuillLMS/client/app/bundles/Proofreader/components/proofreaderActivities/container.tsx

```diff
@@ -300,7 +301,21 @@ export class PlayProofreaderContainer extends React.Component<PlayProofreaderCon
       }
     }
 
+    initializeSubscription(activitySessionUid) {
+      if (process.env.NODE_ENV === 'development') {
+        Pusher.logToConsole = true;
+      }
+      if (!window.pusher) {
+        window.pusher = new Pusher(process.env.PUSHER_KEY, { cluster: process.env.PUSHER_CLUSTER });
+      }
+      const channel = window.pusher.subscribe(activitySessionUid);
+      channel.bind('concept-results-saved', () => {
+        document.location.href = `${process.env.DEFAULT_URL}/activity_sessions/${activitySessionUid}`;
```

Comment:
I realize this is refactoring, but do you know why `this.setState({ saved: true, });`
isn't used here but is with Connect and Grammar?

Reply:
@brendanshean this is a little bit circular, but that line exists in Connect and Grammar but not in Proofreader because there is no reference to a `saved` state value in Proofreader, whereas in Grammar we use it for some business logic in `componentDidUpdate` and in Connect it looks like there's a display state dependent on it.